### PR TITLE
offset allowed

### PR DESCRIPTION
**Pull request by rickyshrestha**
resolve issue - https://github.com/utilitywarehouse/go-pubsub/issues/37

### CODE REVIEW

File: natss/source.go [hunk 1]
```diff
@@ -27,15 +28,44 @@ type MessageSourceConfig struct {
...
+	natsURL             string
+	clusterID           string
+	consumerID          string
+	topic               string
+	offset              Offset
+	offsetStartAtIndex  uint64
+	offsetStartDuration time.Duration
 }
 
+const (
```
Comment (catalin-ilea) [inline natss/source.go]:
```
const (
	//OffsetStartAt - to start at a given msg number"
	OffsetStartAt = Offset(iota)

	//OffsetDeliverLast - deliver from the last message
	OffsetDeliverLast

	//OffsetDeliverAll - deliver all the messages form the begining
	OffsetDeliverAll

	//OffsetStartTime - deliver messages from a certain time
	OffsetStartTime
)
```

File: natss/source.go [hunk 2]
```diff
@@ -27,15 +28,44 @@ type MessageSourceConfig struct {
...
 type messageSource struct {
-	natsURL    string
-	clusterID  string
-	consumerID string
-	topic      string
+	natsURL             string
+	clusterID           string
+	consumerID          string
+	topic               string
+	offset              Offset
```
Comment (catalin-ilea) [inline natss/source.go]: Where are you passing these in?

Comment (catalin-ilea) [reply natss/source.go]:
```
func NewMessageSource(conf MessageSourceConfig) (pubsub.MessageSource, error) {
	return &messageSource{
		natsURL:    conf.NatsURL,
		clusterID:  conf.ClusterID,
		consumerID: conf.ConsumerID,
		topic:      conf.Topic,
	}, nil
}
```

Comment (rickyshrestha) [reply natss/source.go]: Yeah, that was not being passed into the MessageSource. Fixed it.

File: natss/source.go [hunk 3]
```diff
@@ -77,8 +107,24 @@ func (mq *messageSource) ConsumeMessages(ctx context.Context, handler pubsub.Con
...
+	case OffsetDeliverAll:
+		startOpt = stan.DeliverAllAvailable()
+
+	case OffsetStartTime:
+		startOpt = stan.StartAtTimeDelta(mq.offsetStartDuration)
+	}
+
 	subcription, err := conn.QueueSubscribe(mq.topic, mq.consumerID, f, startOpt, stan.DurableName(mq.consumerID), stan.SetManualAckMode())
 
 	defer subcription.Close()
```
Comment (catalin-ilea) [inline natss/source.go]: You should move this after the err check otherwise it will panic on err.

Comment (rickyshrestha) [reply natss/source.go]: done.

**Pull request merged**